server/data/data-api/src/main/java/org/apache/james/droplists/api/DropListEntry.java

```diff
@@ -0,0 +1,122 @@
+ * under the License.                                           *
+ ****************************************************************/
+package org.apache.james.droplists.api;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import com.google.common.base.Preconditions;
+
+public class DropListEntry {
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public static class Builder {
+        private Optional<OwnerScope> ownerScope = Optional.empty();
+        private Optional<String> owner = Optional.empty();
+        private Optional<String> deniedEntityType = Optional.empty();
+        private Optional<String> deniedEntity = Optional.empty();
+
+        public Builder ownerScope(OwnerScope ownerScope) {
+            Preconditions.checkNotNull(ownerScope);
+            this.ownerScope = Optional.of(ownerScope);
+            return this;
+        }
+
+        public Builder owner(String owner) {
+            Preconditions.checkNotNull(owner);
+            this.owner = Optional.of(owner);
+            return this;
+        }
+
+        public Builder deniedEntityType(String deniedEntityType) {
+            Preconditions.checkNotNull(deniedEntityType);
+            this.deniedEntityType = Optional.of(deniedEntityType);
+            return this;
+        }
+
+        public Builder deniedEntity(String deniedEntity) {
+            Preconditions.checkNotNull(deniedEntity);
+            this.deniedEntity = Optional.of(deniedEntity);
+            return this;
+        }
+
+        public Builder copy(DropListEntry dropListEntry) {
+            this.ownerScope = dropListEntry.getOwnerScope();
+            this.owner = dropListEntry.getOwner();
+            this.deniedEntityType = dropListEntry.getDeniedEntityType();
+            this.deniedEntity = dropListEntry.getDeniedEntity();
+            return this;
+        }
+
+        public DropListEntry build() {
+            return new DropListEntry(ownerScope, owner, deniedEntityType, deniedEntity);
+        }
+    }
+
+    private Optional<OwnerScope> ownerScope;
+    private Optional<String> owner;
+    private Optional<String> deniedEntityType;
+    private Optional<String> deniedEntity;
+
+    public DropListEntry(Optional<OwnerScope> ownerScope, Optional<String> owner, Optional<String> deniedEntityType, Optional<String> deniedEntity) {
+        this.ownerScope = ownerScope;
+        this.owner = owner;
+        this.deniedEntityType = deniedEntityType;
+        this.deniedEntity = deniedEntity;
+    }
+
+    public Optional<OwnerScope> getOwnerScope() {
+        return ownerScope;
+    }
+
+    public Optional<String> getOwner() {
+        return owner;
+    }
+
+    public Optional<String> getDeniedEntityType() {
+        return deniedEntityType;
+    }
+
+    public Optional<String> getDeniedEntity() {
+        return deniedEntity;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
```

Comment:
Please write a POJO test with EqualsVelifier. I bet this POJO is non-compliant...